assemblers/org.osbuild.qemu

```diff
@@ -136,10 +136,23 @@ def main(tree, output_dir, options, loop_client):
     extra_args = []
 
     # raw and vdi don't suppport compression
-    if fmt not in ("raw", "vdi"):
+    if fmt not in ("raw", "vdi", "vpc"):
         extra_args.append("-c")
 
-    subprocess.run(["qemu-img", "convert", "-O", fmt, *extra_args, image, f"{output_dir}/{filename}"], check=True)
+    if fmt != "vpc":
+        subprocess.run(["qemu-img", "convert", "-O", fmt, *extra_args, image, f"{output_dir}/{filename}"], check=True)
+    else:
+        # Implementation of the resize algorithm from here:
+        # https://docs.microsoft.com/en-us/azure/virtual-machines/linux/create-upload-generic
+        mb = 1024*1024
+        rounded_size = int((size/mb + 1) * mb)
+        subprocess.run(["qemu-img", "resize", image, rounded_size], check=True)
+        subprocess.run(["qemu-img", "convert",
+                        "-f", "raw",
+                        "-o", "subformat=fixed,force_size",
```

Comment:
Hm, could we just keep one branch here and only do one call to `qemu-img convert` and instead append to the options in the vpc case above in the same way the `-c` option is appended.

Also, could you include some background/info about these options, I didn't find them in the manpage. Is this this something we only do for vpc, is it something we must do? Could we do it unconditionally?

Reply:
I don't know, it was in the tutorial from Microsoft and I didn't want to introduce it for other image formats so I don't break what already works :)